lib/src/room.dart

```diff
@@ -425,36 +422,61 @@ class KurentoRoom extends ChangeNotifier implements Room {
     }
   }
 
+  Future<void> _getServiceRequestStatus() async {
+    Map<String, String?> response = await _client.getServiceRequestStatus(_serviceRequest.id);
+
+    await _updateServiceRequestStatus(response['status']!, response['agentFirstName']);
+  }
+
   Future<void> _handleServiceRequestPresenceMessage(String message) async {
     Map<String, dynamic> json = jsonDecode(message);
-    if (json['id'] != _serviceRequest.id) {
-      // Ignore status updates for other service requests (right now, Platform enforces that an Explorer can only be in
-      // one session, but that may change in the future).
-      return;
-    }
+    if (json['id'] != _serviceRequest.id) return;
 
-    _log.info('service request status changed id=${json['id']} status=${json['status']}');
+    await _updateServiceRequestStatus(json['status'], json['agentFirstName']);
+  }
+
+  Future<void> _updateServiceRequestStatus(String status, String? agentFirstName) async {
+    _log.info('updating service request status=$status');
+
+    switch (status) {
+      case 'REQUEST':
+        // Nothing to do.
+        break;
 
-    switch (json['status']) {
       case 'ASSIGNED':
-        _serviceRequestState = ServiceRequestState.assigned;
-        _agentName = json['agentFirstName'];
+        if (_serviceRequestState == ServiceRequestState.queued) {
```

Comment:
We now handle invalid state transitions. For example, it's not valid to go from `started` to `assigned`.